### ISSUE TITLE
How should decrypt result responses be formatted?

### ISSUE THREAD

**Issue by Kritner** [usnistgov/ACVP#27]
Since AES-GCM is slated as the first algorithm to be implemented, I'll use it as an example.  Given the current `result` sample JSON from `acvp_sub_symmetric`:
```
{
    "acvVersion": "0.2",
    "vsId": 1564,
    "testResults": [
        {
            "tcId": 2171,
            "iv": "01020304C077A707E56C22AC",
            "ct": "",
            "tag": "53B15E85E377F8B5B6BFC2DE915FD622"
        },
        {
            "tcId": 2172,
            "iv": "01020304D666587803BCBAD7",
            "ct": "",
            "tag": "E7B1F2D77D82C6EB3C9CEFF16C376D9C"
        },
        {
            "tcId": 2173,
            "iv": "01020304D3B09894BB14B6CF",
            "ct": "",
            "tag": "DFF22B329AC87B82726EBBF20E66EEE2"
        },
        {
            "tcId": 2174,
            "iv": "01020304D678FBCC2EA341A2",
            "ct": "",
            "tag": "BAFBFA38AD5D6C6E7C97D08DDEC26580"
        },
        {
            "tcId": 2186,
            "iv": "01020304C6A745B0C4C9F2D5",
            "ct": "",
            "tag": "429FD5A0DAB344EF69B594A44CCAD3EB"
        },
        {
            "tcId": 2187,
            "iv": "010203043E56946A9C9E18D7",
            "ct": "",
            "tag": "7DA8BA92D64071909972395C981C4567"
        },
        {
            "tcId": 2201,
            "iv": "0102030479FD178FEB555D99",
            "ct": "FA2AF4B01BA610DE3A3EDE0B88946E2D",
            "tag": "3E77B921DB447F541EB5CC508D74CE4E"
        },
        {
            "tcId": 2202,
            "iv": "01020304C70FB9D3B9E1E329",
            "ct": "568028D34EA82E8D30D1A3B952849A70",
            "tag": "3D20D5935D6D37F6EE82FE0D2C7006F8"
        },
        {
            "tcId": 2203,
            "iv": "010203040D66A6EEA797B757",
            "ct": "B2E10E2ECEC9FC3477841D3AED6469DB",
            "tag": "DEF33A719333CAADE8002752922EC124"
        },
        {
            "tcId": 2204,
            "iv": "010203045B6CDA4B5CA26883",
            "ct": "CF72C75B4E550AEA535750D6B16393DF",
            "tag": "370D1175CF36A72585F20B379D064BCF"
        },
        {
            "tcId": 2205,
            "iv": "010203041E9FD60F49957FF5",
            "ct": "A1F124EE6E98CAB776CCD601B7F65D6A",
            "tag": "F4D7D0D75FC9961A12B3C6A3E6379972"
        },
        {
            "tcId": 2206,
            "iv": "0102030474CBC83C647DBEAF",
            "ct": "2CE1129D3E2F25468A9EE13ED39C902C",
            "tag": "48F3D4D7EB4B9FCB924ACC05B3834F98"
        }
    ]
}
```

In above, it appears that all samples are "encrypt" samples, however "decrypt" test vectors/test cases will also be generated.  

Decrypt test case possibility:

```
{
    "tcId": 9999,
    "key": "FEFFE9928665731C6D6A8F9467308308",
    "iv": "CAFEBABEFACEDBADDECAF888",
    "ct": "42831EC2217774244B7221B784D0D49CE3AA212F2C02A4E035C17E2329ACA12E21D514B25466931C7D8F6A5AAC84AA051BA30B396A0AAC973D58E091473F5985",
    "aad": "3AD77BB40D7A3660A89ECAF32466EF97F5D3D58503B9699DE785895A96FDBAAF43B1CD7F598ECE23881B00E3ED0306887B0C785E27E8AD3F8223207104725DD4",
    "tag": "64C0232904AF398A5B67C10B53A5024D",  
},
```
I would think the `result` for a decrypt test case would look something like:

```
{
    "tcId": 9999,
    "iv": "CAFEBABEFACEDBADDECAF888",
    "pt": "D9313225F88406E5A55909C5AFF5269A86A7A9531534F7DA2E4C303D8A318A721C3C0C95956809532FCF0E2449A6B525B16AEDF5AA0DE657BA637B391AAFD255",
    "tag": "64C0232904AF398A5B67C10B53A5024D"
}
```

Where `ct` (cipherText) is replaced with `pt` (plainText).  Both "iv" and "tag" are not actually needed in a decrypt operation result, so those could potentially be removed.

Thoughts?

**Comment by bfussell** [comment]
Your decrypt request example seems accurate.

Regarding your last point, I agree. The thought for ACVP was that we would not carry all the data from the request back with the response.  This provides a bit of secrecy in that no one can look at the response information and tell what the request vectors were.  With that in mind for GCM decrypt(or any of the AES decrypts) only the PT is returned. There is no need to carry the IV or TAG in the decrypt result.

The failure decrypt results could simply return:

"pt" : "FAIL"

or if we want a separate keyword for the failing case:

"result" : "FAIL"

**Comment by Kritner** [comment]
Thanks @bfussell I'll take that into account for dropping `IV` and `tag` from decryption results.  I'll copy over your response regarding the other issue in #28   I guess keep the issue open until it's worked into the spec?

**Comment by atvassilev** [comment]
Addressed by PR  #36
